src/parser.rs

```diff
@@ -154,26 +124,24 @@ fn collapse_chunks(chunks: &[EncodedChunk]) -> Result<Vec<EncodedChunk>, Restore
 
     let mut mismatching = Vec::<(EncodedChunk, String)>::new();
     // Group by chunk id
-    for (_chunk_id, duplicate_chunks) in &chunks.into_iter().group_by(|chunk| chunk.id) {
-        let mut duplicate_chunks_iter = duplicate_chunks.into_iter();
-        let reference_chunk = match duplicate_chunks_iter.next() {
+    for (_chunk_id, mut duplicate_chunks) in &chunks.iter().group_by(|chunk| chunk.id) {
+        let reference_chunk = match duplicate_chunks.next() {
```

Comment:
So by not calling duplicate_chunks iterable, I made its iterator-ness be unavailable due to oversight?
Thanks for solving this, the code definitely was uglier because of the trick I had to pull.